diff --git a/crypto/algif_aead.c b/crypto/algif_aead.c
index a3be734..b24cd19 100644
--- a/crypto/algif_aead.c
+++ b/crypto/algif_aead.c
@@ -191,7 +191,7 @@ static int _aead_recvmsg(struct socket *sock, struct msghdr *msg,
 
 	/* Activar operacion out-of-place para proteger Page Cache */
-	aead_request_set_crypt(req, rsgl.src, rsgl.src, used, iv);
+	aead_request_set_crypt(req, tsgl.src, rsgl.src, used, iv);
 	err = crypto_aead_decrypt(req);
 	return err;
